### PR TITLE
[jit] Clear cfg->error after inlining, the error can be ignored, it will happen when the called method is JITted.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by jaykrell** [comment]
I would suggest cfg_error_cleanup_reuse or cfg_error_cleanup_reinit or cfg_error_cleanup_init or maybe just cfg_error_cleanup -- with the implicit convention that cfg reuses, but up to you. I like the word "cleanup" instead of "clear" as well, but "clear" is shorter, granted. (I don't like free because it sounds too much like memory-specific and "destroy" and "destructor" are too destructive..)

**Comment by vargaz** [comment]
Clearing an error is the standard terminology for this imho.

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -6113,6 +6120,8 @@ inline_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig,
 	cfg->disable_inline = prev_disable_inline;
 	cfg->inline_depth --;
 
+	clear_cfg_error (cfg);
```
Comment (lambdageek) [inline mono/mini/method-to-ir.c]: This is alright, but maybe we could push the error handling down into the two branches of the `if`, below, and try to get something out of the error:

1. If we decide to inline, can't we check that `cfg->error` must have been ok?
   ```c
 		if (cfg->verbose_level > 2)
  			printf ("INLINE END %s -> %s\n", mono_method_full_name (cfg->method, TRUE), mono_method_full_name (cmethod, TRUE));
   ```
2. If we find out we can't inline, can't we check if the error is not ok and print it out if the verbosity is high enough?
   ```c
  		if (cfg->verbose_level > 2)
 			printf ("INLINE ABORTED %s (cost %d)\n", mono_method_full_name (cmethod, TRUE), costs);
   ```

Comment (vargaz) [reply mono/mini/method-to-ir.c]: If the error is set method_to_ir () returns -1 i.e. an error, so the inline path won't happen.

Comment (lambdageek) [reply mono/mini/method-to-ir.c]: Could it happen that `inline_always` is true but `costs == -1` and `cfg->error` is set?

**Comment by jaykrell** [comment]
Between memory and files and sockets and errors and all resources, I like to try to have just one word for the release of the resource. "free" seems too memory-centric maybe, "close" is too file-centric maybe. "cleanup" feels like one word that might semi-reasonably apply to all resources. Maybe it is stretch, because, e.g. lock_cleanup to mean unlock? Maybe "release". Cleanup sounds kinda informal, but ok to me. Release sounds kinda formal and I hadn't really considered it. It is the name chosen for COM to decrement a reference count, so that is reasonable precedent. "clear" doesn't seem to fit other resources, though again I hadn't considered it. Maybe it isn't possible and just live with multiple names for it.  esp. in existing codebase. For example maybe a lock can both be unlocked and freed, maybe it holds onto an underlying resource as long as it remains lockable.

C++ gives us "destructor" with the short spelling "~", and automatically called at that but I don't really like the word destructor or destroy and we can't use "~" that way ~error(&cfg->error).

File: mono/mini/method-to-ir.c [hunk 2]
```diff
@@ -6117,6 +6124,8 @@ inline_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig,
 		if (cfg->verbose_level > 2)
 			printf ("INLINE END %s -> %s\n", mono_method_full_name (cfg->method, TRUE), mono_method_full_name (cmethod, TRUE));
 
+		g_assert (is_ok (&cfg->error));
```
Comment (luhenry) [inline mono/mini/method-to-ir.c]: Use `mono_error_assert_ok`, it does the same but with a message that gives more info.

**Pull request merged**